admin/class-yoast-dashboard-widget.php

```diff
@@ -74,12 +74,41 @@ public function clear_cache() {
 	 * @return array
 	 */
 	private function statistic_items() {
+		$transient = get_transient( self::CACHE_TRANSIENT_KEY );
 
-		if ( false !== ( $items = get_transient( self::CACHE_TRANSIENT_KEY ) ) ) {
-			return $items;
+		if ( isset( $transient[$this->get_user_id()][1] ) ) {
+			return $transient[$this->get_user_id()];
 		}
 
-		$items = array(
+		return $this->set_statistic_items_for_this_user( $transient );
+	}
+
+	/**
+	 * Set the cache for a specific user
+	 *
+	 * @param $transient
+	 *
+	 * @return mixed
+	 */
+	private function set_statistic_items_for_this_user( $transient ) {
+		if ( $transient === false ) {
+			$transient = array();
+		}
+
+		$filtered_items[$this->get_user_id()] = array_filter( $this->get_seo_scores_with_post_count(), array( $this, 'filter_items' ) );
```

Comment:
I would store the user id into it's own variable, because this won't change during the method load.